fbw-common/src/wasm/systems/systems/src/air_conditioning/mod.rs

```diff
@@ -236,17 +236,17 @@ impl From<usize> for Channel {
 
 struct OperatingChannel {
     channel_id: Channel,
-    powered_by: ElectricalBusType,
+    powered_by: Vec<ElectricalBusType>,
     is_powered: bool,
     failure: Failure,
     fault: OperatingChannelFault,
 }
 
 impl OperatingChannel {
-    fn new(id: usize, failure_type: FailureType, powered_by: ElectricalBusType) -> Self {
+    fn new(id: usize, failure_type: FailureType, powered_by: &[ElectricalBusType]) -> Self {
         Self {
             channel_id: id.into(),
-            powered_by,
+            powered_by: powered_by.to_vec(),
```

Comment:
I don't see why we need a vector here
```suggestion
    powered_by: [ElectricalBusType; 2],
    is_powered: bool,
    failure: Failure,
    fault: OperatingChannelFault,
}

impl OperatingChannel {
    fn new(id: usize, failure_type: FailureType, powered_by: [ElectricalBusType; 2]) -> Self {
        Self {
            channel_id: id.into(),
            powered_by,
```

Reply:
In this case each ACSC lane is powered by 2 electrical buses, but I want to be able to use this struct on a380 systems (FDAC, TADD, VCM, OCSM) - see #8086. Some of them are powered by a single bus per channel. I want to keep it generic so a vector is the best way I found to be able to do this, unless you know of a better way!

Reply:
Ok, then I think a vector is fine. Another possibility would be to introduce const generics.